### PR TITLE
core: deprecate ModuleOp.from_region_or_ops

### PR DESCRIPTION
**Pull request by superlopuh**
I thought I'd deprecate instead of deleting since it's the most widely-used Operation, but I'd be ok with deleting the constructor also.

### CODE REVIEW

File: tests/interpreters/test_pdl_interpreter.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def test_rewrite_swap_inputs_pdl():
 
 def swap_arguments_input():
 
-    @ModuleOp.from_region_or_ops
+    @ModuleOp
```
Comment (superlopuh) [inline tests/interpreters/test_pdl_interpreter.py:57]: 👨‍🍳

**Review by webmiche — APPROVED** [review]
I am fine with deleting as well. Personally, I don't think the API is supposed to be stable at all before a 1.0 release, therefore, I don't think we need to involve deprecation messages. But I know there are other opinions here, so I am fine with both :)

**Comment by superlopuh** [comment]
If one more person tells me to delete it I will :)

**Comment by tobiasgrosser** [comment]
I am fine with deleting deprecation things. I feel we are still polishing the API, so we do not necessarily need them. In fact, there is a higher risk that people start to use deprecated APIs than them being in trouble of loosing an API.

**Pull request merged**